src/common/data-store/index.js

```diff
@@ -0,0 +1,95 @@
+import browser from "webextension-polyfill";
+
+import { encryptData, decryptData } from "../../common/lib/crypto";
+
+/**
+ * Encripted storage module. A passord is required to initialize this module.
+ * The storage is under the form of key-value pairs.
+ */
+let cachedStorage = null;
+
+function init(password) {
+  if (!password) {
+    throw new Error("A passord is required to initialize the storage module!");
+  }
+
+  const pwd = `${password}`;
+
+  function get(key) {
+    if (!key || typeof key !== "string") {
+      throw new Error("Invalid key: " + key);
+    }
+    return browser.storage.sync.get(key).then((result) => {
+      try {
+        if (!result || !result[key]) return;
+        const decriptedData = decryptData(result[key], pwd);
+        return decriptedData && decriptedData.data;
+      } catch (err) {
+        console.error(err);
+        throw new Error("Cannot decode data from storage. Wrong password!");
+      }
+    });
+  }
+
+  function set(key, data) {
+    if (!key || typeof key !== "string") {
+      throw new Error("Invalid key: " + key);
+    }
+
+    const encryptedData = encryptData({ data }, pwd);
+    const kV = {
+      [key]: encryptedData,
+    };
+    return browser.storage.sync.set(kV);
+  }
+
+  return {
+    set,
+    get,
+    isInitialized: async function () {
+      // TODO: recheck
+      const isPasswordSet = await browser.storage.sync.get("isPasswordSet");
+      return (
+        isPasswordSet !== undefined &&
+        JSON.stringify(isPasswordSet) !== JSON.stringify({})
+      );
+    },
+    isUnlocked: async function () {
+      return true;
+    },
+  };
+}
+
+function initNullDataStore() {
```

Comment:
why do we need the null data store that throws errors? 
can't we prevent accessing some kind of storage and not have a store at all if it is not decrypted? (besides that I think we need to discuss what to encrypt as mentioned above)